src/plugins/workspace/public/components/data_source_association/association_data_source_modal.tsx

```diff
@@ -231,26 +255,53 @@ export const AssociationDataSourceModalContent = ({
   useEffect(() => {
     setIsLoading(true);
     getDataSourcesList(savedObjects.client, ['*'])
-      .then((dataSourcesList) => fetchDataSourceConnections(dataSourcesList, http, notifications))
-      .then((connections) => {
-        setAllConnections(connections);
+      .then((dataSourcesList) => {
+        const { openSearchConnections, dataConnections } = getOpenSearchAndDataConnections(
+          dataSourcesList
+        );
+        setAllConnections([...openSearchConnections, ...dataConnections]);
+        return { openSearchConnections, dataConnections, dataSourcesList };
+      })
+      .then(({ openSearchConnections, dataConnections, dataSourcesList }) => {
+        fetchDirectQueryConnections(dataSourcesList, http, notifications).then(
```

Comment:
Nit: I'm thinking of do we need to have a loading status for direct query connections for each data source. Since load direct query connections may take long time, but if we separate to each data source, it would be more faster.

Reply:
Thanks for the advice, updated and new screenshot is: 

Uploading 录屏2024-12-11 17.46.05.mov…

Reply:
Yes, maybe you can call `getDirectQueryConnections` to get the DQCs separately and set the isRelatedConnectionsLoaded status according to the data source id.